packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -440,8 +440,6 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
       visibleRows: currentPage.rows,
     });
 
-    const renderedColumns = visibleColumns.slice(firstColumnToRender, lastColumnToRender);
```

Comment:
Instead of memoizing `renderedColumns`, I decided to not pass them to `GridRow`.
Since `visibleColumns`, `firstColumnToRender` and `lastColumnToRender` are passed to `GridRow` anyway, and it's enough to replicate `renderedColumns`